src/sentry/constants.py

```diff
@@ -549,6 +549,7 @@ def from_str(cls, string: str) -> Optional[int]:
 INTERNAL_INTEGRATION_TOKEN_COUNT_MAX = 20
 
 ALL_ACCESS_PROJECTS = {-1}
+ALL_ACCESS_PROJECTS_SLUG = {"$all"}
```

Comment:
@nhsiehgit Would love a test for this. 

Is `$all` a project slug that's impossible to set?

Reply:
Yessir - sorry i have a habit of getting the code out first, then working on tests next 😬 
working on that now.

@dashed - yes, slugs starting in `$` are invalid. (as well as other special chars i believe)

Reply:
@nhsiehgit its probably worth adding it to `RESERVED_PROJECT_SLUGS` just in case in 10 years we change how slugs are validated or something :P